cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -306,6 +311,11 @@ func (c *CStorVolumeReplicaController) cVRAddEventHandler(
 			string(common.AlreadyPresent),
 			string(common.MessageResourceAlreadyPresent),
 		)
+		// If user has perform upgrade then replicaID may not be set
+		if err := volumereplica.UpdateReplicaID(cVR); err != nil {
```

Comment:
No need to handle upgrade is already handling it.

Reply:
yes. I have added here if upgrade/user missed to set replicaID. It will check if replicaID is set or not, if not set then it will update else return nil.